Source/carthage/Outdated.swift

```diff
@@ -54,13 +55,40 @@ public struct OutdatedCommand: CommandProtocol {
 
 				if !outdatedDependencies.isEmpty {
 					carthage.println(formatting.path("The following dependencies are outdated:"))
-					for (project, current, updated) in outdatedDependencies {
+
+					for (project, current, updated, latest) in outdatedDependencies {
+						let versionColor: Color.Named.Color
+						switch (current, updated, latest) {
+						case (_, updated, latest) where updated == latest:
+							// Update available and applicable
+							versionColor = .green
+						case (current, updated, latest) where current != updated && updated != latest:
+							// Update availabe and applicable, but not to the latest version due to version lock
+							versionColor = .yellow
+						case (current, updated, latest) where current == updated:
+							// Update available, but not applicable due to version lock
+							versionColor = .red
+						default:
+							versionColor = .white
+						}
+
 						if options.outputXcodeWarnings {
-							carthage.println("warning: \(formatting.projectName(project.name)) is out of date (\(current) -> \(updated))")
+							carthage.println("warning: \(formatting.projectName(project.name)) is out of date (\(current) -> \(updated) (Latest: \(latest))_")
 						} else {
-							carthage.println(formatting.projectName(project.name) + " \(current) -> \(updated)")
+							let versionSummary = formatting.colored(current.description, color: versionColor)
+								+ " -> " + formatting.colored(updated.description, color: versionColor)
+								+ " (Latest: \(latest))"
+							carthage.println(formatting.projectName(project.name) + " " + versionSummary)
 						}
 					}
+
+					if options.colorOptions.formatting.isColorful {
+						carthage.println(formatting.path("The color indicates what happens when you run `carthage update`"))
+						carthage.println(formatting.colored("<green>", color: .green) + "\t\t- Will be updated to the newest version")
+						carthage.println(formatting.colored("<yellow>", color: .yellow) + "\t- Will be updated, but not to the newest version"
+							+ "because of the specified version in Cartfile")
+						carthage.println(formatting.colored("<red>", color: .red) + "\t\t- Will not be updated because of the specified version in Cartfile")
```

Comment:
This seems like it should be specified in the help instead of being included in every invocation.

Reply:
I'm unsure how to do that. The simple solution I see is to add the text to
`public let function = "Check for compatible updates to the project's dependencies"`, then it will be printed at the top. But that seems wrong to me.

I'd like it much better to include the explanation to the help text of the color option, e.g.:

```
[--color (color)]
	whether to apply color and terminal formatting (one of 'auto', 'always', or 'never')
	
	The color indicates what happens when you run `carthage update`
	<green>	- Will be updated to the newest version
	<yellow>	- Will be updated, but not to the newest version
	<red>	- Will not be updated because of the specified version in Cartfile
```

However, I have no Idea how to implement this since the color option help is generated by `ColorOptions.evaluate(mode)` and I don't see a way to add extra text to the help of `ColorOptions`...